pkgs/applications/graphics/darktable/default.nix

```diff
@@ -37,14 +37,17 @@ stdenv.mkDerivation rec {
   # darktable changed its rpath handling in commit
   # 83c70b876af6484506901e6b381304ae0d073d3c and as a result the
   # binaries can't find libdarktable.so, so change LD_LIBRARY_PATH in
-  # the wrappers:
+  # the wrappers.
+  # We also need to add a symlink to the shared dir containing the
+  # lensfun database:
   preFixup = let
     libPathEnvVar = if stdenv.isDarwin then "DYLD_LIBRARY_PATH" else "LD_LIBRARY_PATH";
     libPathPrefix = "$out/lib/darktable" + stdenv.lib.optionalString stdenv.isLinux ":${ocl-icd}/lib";
   in ''
     gappsWrapperArgs+=(
       --prefix ${libPathEnvVar} ":" "${libPathPrefix}"
     )
+    ln -s ${lensfun}/share/lensfun $out/share
```

Comment:
Instead of having lensefun a dependency is it possible to point darktable to the directory via an environment variable? We could then just have a wrapper around darktable setting that instead of triggering a rebuild of darktable when lensefun changes.

Reply:
It would be nice to avoid rebuilds, but:

* It does not seem like there is an env variable to use (based on [this](https://lensfun.github.io/manual/latest/dbsearch.html))
* `lensfun` is [already one of the `buildInputs` of `darktable`](https://github.com/NixOS/nixpkgs/blob/c61ae45b08b1b21735daae2eb7b0bcb7e5f0a3da/pkgs/applications/graphics/darktable/default.nix#L20-L21) (so it already triggers a rebuild)

Reply:
Would not it be preferable to patch the path here: https://github.com/darktable-org/darktable/blob/d9d279a1e1b38baaab07981dfa6bc87fe16238cf/src/iop/lens.cc#L1259

Or open an issue to respect `XDG_DATA_DIRS`?

Reply:
Oh.  Interesting.  I had not looked in there.  I am on macOS so it might be that the only problem is that the lensfun `Load` function is not actually being tried first if `__APPLE__` is defined https://github.com/darktable-org/darktable/blob/d9d279a1e1b38baaab07981dfa6bc87fe16238cf/src/iop/lens.cc#L1246-L1249

Reply:
Indeed, on Linux, the correct system path should be hopefully tried: https://github.com/lensfun/lensfun/blob/b58ac41f0f62000acdf6e73e0b5ef7c9af72efef/libs/lensfun/database.cpp#L29

Maybe patching it to use `Load` on macOS too would be enough.

Reply:
I don't see this issue on Linux. I'm not sure if darktable's cmake detects the platform as macOS and tries to build everything into one bundle, as masOS expects it, but this fix sholdn't be necessary on linux.